setpass.py

```diff
@@ -0,0 +1,52 @@
+#   Copyright 2016 Massachusetts Open Cloud
+#
+#   Licensed under the Apache License, Version 2.0 (the "License"); you may
+#   not use this file except in compliance with the License. You may obtain
+#   a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+#   WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+#   License for the specific language governing permissions and limitations
+#   under the License.
+import string
+import random
+
+
+def random_password(size):
+    """Generate a random password of length 'size'
+    
+    The resulting password may contain any of:
+        upper or lowercase letters A-Z
+        the digits 0-9
+        valid punctuation marks defined in the 'punctuation' variable below
+    """
+    punctuation = '#$%&!'
+    chars = string.ascii_letters + string.digits + punctuation
+    return ''.join(random.choice(chars) for _ in range(size))
```

Comment:
Another way to do this is to use a secure hash - SHA-2.  This is the more typical (secure) method that I have seen other people use.  

Chances are random is not really random, but this probably not a problem yet.